static/app/views/organizationContextContainer.tsx

```diff
@@ -16,329 +14,213 @@ import {ORGANIZATION_FETCH_ERROR_TYPES} from 'sentry/constants';
-    const {organization, error} = orgData;
+  // Take a measurement for when organization details are done loading and the
+  // new state is applied
+  useEffect(
+    () => {
+      if (organization === null) {
+        return;
+      }
 
-    if (organization && !error) {
-      // Configure scope to have organization tag
-      Sentry.configureScope(scope => {
-        // XXX(dcramer): this is duplicated in sdk.py on the backend
-        scope.setTag('organization', organization.id);
-        scope.setTag('organization.slug', organization.slug);
-        scope.setContext('organization', {id: organization.id, slug: organization.slug});
+      metric.measure({
+        name: 'app.component.perf',
+        start: 'organization-details-fetch-start',
+        data: {
+          name: 'org-details',
+          route: getRouteStringFromRoutes(routes),
+          organization_id: parseInt(organization.id, 10),
+        },
       });
-    } else if (error) {
-      // If user is superuser, open sudo window
-      const user = ConfigStore.get('user');
-      if (!user || !user.isSuperuser || error.status !== 403) {
-        // This `catch` can swallow up errors in development (and tests)
-        // So let's log them. This may create some noise, especially the test case where
-        // we specifically test this branch
-        console.error(error); // eslint-disable-line no-console
-      } else {
-        openSudo({
-          retryRequest: () => Promise.resolve(this.fetchData()),
-          isSuperuser: true,
-          needsReload: true,
-        });
-      }
+    },
+    // Ignore the `routes` dependency for the metrics measurement
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [organization]
+  );
+
+  // Configure sentry SDK scope to have organization tag
+  useEffect(() => {
+    if (organization === null || error) {
+      return;
     }
 
-    this.setState({...orgData}, () => {
-      // Take a measurement for when organization details are done loading and the new state is applied
-      if (organization) {
-        metric.measure({
-          name: 'app.component.perf',
-          start: 'organization-details-fetch-start',
-          data: {
-            name: 'org-details',
-            route: getRouteStringFromRoutes(this.props.routes),
-            organization_id: parseInt(organization.id, 10),
-          },
-        });
-      }
+    Sentry.configureScope(scope => {
+      // XXX(dcramer): this is duplicated in sdk.py on the backend
+      scope.setTag('organization', organization.id);
+      scope.setTag('organization.slug', organization.slug);
+      scope.setContext('organization', {id: organization.id, slug: organization.slug});
     });
-  }
+  }, [organization, error]);
 
-  renderSidebar(): React.ReactNode {
-    if (!this.props.includeSidebar) {
-      return null;
-    }
-
-    return <Sidebar organization={this.state.organization ?? undefined} />;
-  }
+  const {user} = configStore;
```

Comment:
user is only accessed once, so I would remove this. If it's here because react hooks complains of the reference then so be it and lets leave it as is.